src/utils/PersonIdentityUtils.ts

```diff
@@ -21,22 +20,28 @@ export const personIdentityUtils = {
 		const { givenNames, familyNames } = this.getNames(personDetails);
 		const f2fGivenNames = givenNames.join(" ");
 		const f2fFamilyName = familyNames.join(" ");
-
+    
 		// Check if the fullName in f2f matches the fullName in DocumentFields
 		const doesFullNameMatch = `${f2fGivenNames.toLowerCase()} ${f2fFamilyName.toLowerCase()}` === yotiFullName.toLowerCase();
 		if (!doesFullNameMatch) {
 			const errorMessage = "FullName mismatch between F2F & YOTI";
 			logger.error({ message: errorMessage }, { messageCode: MessageCodes.VENDOR_SESSION_NAME_MISMATCH });
 			throw new AppError(HttpCodesEnum.SERVER_ERROR, errorMessage);
 		}
-
-		// Remove family name from the full name and split at spaces
-		const yotiGivenNameParts = yotiFullName.replace(new RegExp(f2fFamilyName, "i"), "").match(/\S+/g);
-		// Map the array of given names into the correct format
-		const nameParts = yotiGivenNameParts.map((name: string) => ({ value: name.trim(), type: "GivenName" }));
-		// Remove the given names from the full name, remove surrounding spaces, and map to correct format
-		nameParts.push({ value: yotiFullName.replace(new RegExp(f2fGivenNames, "i"), "").trim(), type: "FamilyName" });
-
+    
+		// Split the full name into words
+		const fullNameWords = yotiFullName.split(" ").filter((part: string) => part !== "");
+    
+		// Split the words into given names and family name parts
+		const nameParts = [];
+		for (const word of fullNameWords) {
```

Comment:
I might be being a bit annoying here but what if the name was `boat mcboatface boaty`? Then the includes check would pass on both `boaty` and `boat` so we'd end up with this, no?

<img width="625" alt="image" src="https://github.com/alphagov/di-ipv-cri-f2f-api/assets/40401118/e8b03771-cf75-4de2-89d6-56c16fd534d3">

Reply:
Yeah you're right, back to the drawing board on this one